src/emitter/emitter.ts

```diff
@@ -31,7 +31,7 @@ export class EventEmitter<E extends EventMap> implements IEventEmitter<E> {
       registeredListeners = new WeakSet();
       this._registeredListeners.set(type, registeredListeners);
     } else if (registeredListeners.has(listener)) {
-      return createDisposable(Function);
+      throw new Error(`Double-registered for '${type}' event.`);
```

Comment:
Well, I'm a bit on the edge with this one.
While I agree this would be perfect to have in, I wanted to keep the PR relatively safe. :grin: